### DISCUSSION TITLE
Changing defaults

### DISCUSSION THREAD

**Discussion by Kaedriz** [soulhotel/FF-ULTIMA#20] [General]
@soulhotel, what do you think about changing the current defaults? Specifically, z-sites and tabs width. `z-sites.css` should be op-in, not opt-off like it is currently. And medium tab width is not that useful middle ground in my opinion, maybe widening it a bit could add more use to it. Another one would be autohide, but that can be classified as additional functionality, and as it stands for now is fine probably.

For development purposes you can just commit all your personal modification to a separate branch, use that branch, and pull changes from main when the update is out. Then you would just checkout to main for latest default version. That was my original way of applying customization to git way of install, it would preserve modifications regardless of updates, maybe occasional merge would be required to be managed.

**Comment by soulhotel** [comment]
Perfect timing, Yeah I do agree `z-sites` could probably be removed all together, and tab sizes set to `positioning-L` for default... I do want to leave the autohide function disabled, its so easy to change, and I encourage others to open up the files and browse/disable things they don't like.

As for development once i have the time (april 12th) I will definitely set this up.

I also plan on making some corrections for the theme when used on LINUX, should only take an hour or so, and im dual booting a DE now.

Also going to attempt to overhaul the tab system, The biggest issue being the drag and drop functionality, it may just be impossible with only CSS but im gonna try again.

**Reply by Kaedriz** [reply]
As I have looked into it, only horizontal mouse movement is registered, even if you rotate the native tab bar. Dunno if there is anything that can be done here with just CSS. Firefox is working on native vertical tabs, but it will take them awhile, if ever. (There was a prototype vert tab bar in nightly a while back)

**Reply by soulhotel** [reply]
I know! Either way im patiently waiting to see what they can do. And I dont want to deal with the headaches of config scripts. And take your time brother, we all have lifes to get back to.

**Comment by soulhotel** [comment]
I just created a new branch, new-defaults.

- L size tabs `250px` will be the new default
- `z-sites.css` has been disabled
- S size tabs changed from `140px` to `165px`
- From the replace-features list Ive added `position-verttab-gone.css` into the theme folder. Eventually this will be labeled as an optional feature for people who rather use sidebery/whatever instead of the Native Tabs.

i'll wait and see what you think first.

**Reply by Kaedriz** [reply]
Will look into it more specifically later this day, or maybe tomorrow.

**Comment by Kaedriz** [comment]
Well, for now seems to be fine, only regression is windows controls, but I understand focus for Gnome. In the future, we could support 3 styles for Linux:
Gnome, KDE, and tiling (no controls at all). Also noticed buttons on top bar aren't centralized, was it like that before? I have mild OCD on things like that, strive to be pixel perfect. There are some annoyances spread here and there, will write them to issues soon to work on them.

**Reply by soulhotel** [reply]
Hey man how've you been, I dont think there are CSS selectors to filter if the current linux is on gnome or kde. BUT, the new `override-linux.css` should be compatible with both either way, the only difference is that the user may have to add +/- pixels to the `top: 3px` if it is kde. When I first made my fixes for linux in `1.6.5` I was using Manjaro KDE in a VM, this new file has been tested while dual-booting POPOS Gnome, and it was perfect.

for tiling support, they'd have to disable the 'return-titlebar-buttons` file in userchrome.

You said top bar buttons are not centralized, are you talking about the `minimize window button` in the picture below? It annoys me too, but this is how the default button in popos looks.

 ![2024-04-07_01-53](https://github.com/soulhotel/FF-ULTIMA/assets/155501797/7c227735-45b2-4b21-b095-0e0b2c552e5e)

**Reply by Kaedriz** [reply]
No, I'm think more of address bar buttons, they are not centralized vertically I think.

![address_bar](https://github.com/soulhotel/FF-ULTIMA/assets/44523955/7b1ed5cc-8407-4d4b-8361-2fe8030d4dfe)

And with Linux controls I meant just use current pattern, set default to gnome, then offer kde or no tile option. It would be most logical in my mind

**Reply by soulhotel** [reply]
`all-global-theme` files, line 214, line 223, removed:
```
#urlbar[breakout][breakout-extend] {top: 2px !important;}
/#urlbar-container, #search-container {top: 2px !important;}
```
`all-global-positioning` files, line 33-40, removed top margins things such as sidebar, tabs, browser content
```
--uc-all-margins: 0px 6px 6px 6px; 
--uc-tab-margins: 0px 0px 6px 6px;
--uc-sb-margin-leftside: 0px 3px 6px 6px;
--uc-sb-margin-rightside: 0px 6px 6px 0px;
```
result

![2024-04-08_18-31](https://github.com/soulhotel/FF-ULTIMA/assets/155501797/3e02b2a7-cad6-42be-9a36-f1b373dc21cd)

**Reply by Kaedriz** [reply]
Has the address bar widened vertically? In your screen, it looks very squished. I would like to it only affect buttons, and leave address input untouched. But overall good.

**Reply by soulhotel** [reply]
urlbar width is untouched, this is just the fresh install layout of the nav bar

**Reply by soulhotel** [reply]
basically the tabs, browser content, etc had margins at the top, this made it seem like the nav bar was bigger than it was, also making the buttons seem uncentered vertically. removing those margins like i did above fixes it.
